x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/pipeline_select.test.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setMockActions, setMockValues } from '../../../../../__mocks__/kea_logic';
+
+import React from 'react';
+
+import { mount, shallow } from 'enzyme';
+
+import { EuiSelectable } from '@elastic/eui';
+
+import { PipelineSelect } from './pipeline_select';
+
+const DEFAULT_VALUES = {
+  addInferencePipelineModal: {
+    configuration: {
+      pipelineName: '',
+    },
+  },
+  existingInferencePipelines: [],
+};
+
+const MOCK_ACTIONS = {
+  selectExistingPipeline: jest.fn(),
+};
+
+describe('PipelineSelect', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues({});
+    setMockActions(MOCK_ACTIONS);
+  });
+  it('renders pipeline select with no options', () => {
+    setMockValues(DEFAULT_VALUES);
+
+    const wrapper = shallow(<PipelineSelect />);
+    expect(wrapper.find(EuiSelectable)).toHaveLength(1);
+    const selectable = wrapper.find(EuiSelectable);
+    expect(selectable.prop('options')).toEqual([]);
+  });
+  it('limits pipeline select height to option count', () => {
+    setMockValues({
+      ...DEFAULT_VALUES,
+      existingInferencePipelines: [
+        {
+          modelId: 'model_1',
+          modelType: 'model_1_type',
+          pipelineName: 'pipeline_1',
+          sourceFields: [],
+        },
+        {
+          modelId: 'model_2',
+          modelType: 'model_2_type',
+          pipelineName: 'pipeline_2',
+          sourceFields: [],
+        },
+        {
+          modelId: 'model_3',
+          modelType: 'model_3_type',
+          pipelineName: 'pipeline_3',
+          sourceFields: [],
+        },
+      ],
+    });
+
+    const wrapper = mount(<PipelineSelect />);
```

Comment:
I had to use `mount` (instead of `shallow`) for some of these tests to ensure that the methods that dynamically set the selectable height execute